ext/opcache/jit/zend_jit_trace.c

```diff
@@ -7099,6 +7099,7 @@ static zend_jit_trace_stop zend_jit_compile_root_trace(zend_jit_trace_rec *trace
 			    if (t->stack_map_size) {
 					zend_jit_trace_stack *shared_stack_map = (zend_jit_trace_stack*)zend_shared_alloc(t->stack_map_size * sizeof(zend_jit_trace_stack));
 					if (!shared_stack_map) {
+						efree(t->stack_map);
```

Comment:
should not you set it to NULL after ?

Reply:
It's not set to NULL in zend_jit_compile_side_trace()... in other code paths, it is however set to NULL. But I guess those could be removed, because nobody cares - if anywould would care, they would be confused as to whether to free it with `efree()` or not free it because it's already in shared memory. In any case, this code is unclear and confusing.

Reply:
Upon further investigation, `t->stack_map` is not accessed again in this function and I believe that `t` / `zend_jit_traces[ZEND_JIT_TRACE_NUM]` is considered free and will be initialized when reused, so it's probably safe to _not_ set it to NULL.

On the other hand, the performance implications of setting it to NULL are practically 0 here, and this would reduce the confusion / increase consistency with other branches.

Reply:
Meanwhile I have written a draft commit (to be submitted as follow-up for https://github.com/php/php-src/pull/10147) which removes the NULL setters in other branches. I think setting stuff to NULL only adds confusion because it makes the reader believe NULL has a meaning, but here it has none.